Sources/DotLottie/Private/DotLottieManager.swift

```diff
@@ -0,0 +1,194 @@
+    case invalidURL
+    case missingManifestFile
+}
+
+class DotLottieManager {
+    // id : path on disk
+    private var filePaths: [String: URL] = [:]
+    
+    // path on disk
+    private var manifestFilePath: URL?
+
+    // Object representation of manifest
+    private var manifest: ManifestModel?
+
+    private var currentAnimationId: String = ""
+    
+    // If an error occurs whilst loading
+    private var errorMessage: String?
+
+    public func initFromWebUrl(url: URL) async throws {
+        // Fetch the file data
+        let urlBook = try await fetchDotLottieAndWriteToDisk(url: url)
+
+        try loadDotLottiePipeline(urlBook: urlBook)
+    }
+    
+    /// Initiliaze the DotLottieManager from an animation inside the main asset bundle.
+    /// - Parameter assetName: Name of the animation inside the asset bundle.
+    public func initFromBundle(assetName: String) throws {
+        let fileData = try fetchFileFromBundle(animationName: assetName,
+                                               extensionName: "lottie")
+        let urlBook = try writeDotLottieToDisk(dotLottie: fileData, fileName: assetName)
+
+        try loadDotLottiePipeline(urlBook: urlBook)
+    }
+    
+    /// Initialize the DotLottieManager with the dotLottie information already written to disk.
+    /// - Parameter urlBook: A dictionnary containg the id of the animation as key, and its path to disk as URL.
+    private func loadDotLottiePipeline(urlBook: [String:URL]) throws {
+        guard let manifestPath = urlBook["manifest"] else { throw DotLottieManagerErrors.missingManifestFile }
+        
+        // Extract out a ManifestModel
+        let manifestObject = try extractManifest(manifestFilePath: manifestPath)
+        
+        // Set properties
+        self.filePaths = urlBook        
+        self.manifest = manifestObject
+        self.manifestFilePath = urlBook["manifest"]
+        
+        // If theres a default animation set the current animation id to that
+        if let aaId = self.manifest?.activeAnimationId {
+            if containsAnimation(animationId: aaId) {
+                self.currentAnimationId = aaId
+            }
+        } else if let manifest = self.manifest {
+            if let firstAnimation =  manifest.animations.first {
+                self.currentAnimationId = firstAnimation.id
+            }
+        }
+    }
+    
+    /// Check if an animation is inside the loaded dotLottie.
+    /// - Parameter animationId: Id of the animation, available from the manifest.json file of the dotLottie.
+    /// - Returns: <#description#>
+    public func containsAnimation(animationId: String) -> Bool {
+        if filePaths[animationId] != nil {
+            return true
+        }
+        
+        return false
+    }
+    
+    /// Returns the ManifestModel of the manifest file.
+    /// - Returns: ManifestModel object of the manifest file.
+    public func getManifest() -> ManifestModel? {
```

Comment:
what would be the value of having this function instead of having the manifest object as public? We don't want the client to be able to change it?

if we don't want to change from outside, can do:

public private(set) var  ...